source/egg/core/Archive.cc

```diff
@@ -16,41 +18,55 @@ s32 Archive::convertPathToEntryId(const char *path) const {
     return m_handle.convertPathToEntryId(path);
 }
 
-void *Archive::getFile(s32 entryId, Abstract::ArchiveHandle::FileInfo &info) const {
+void *Archive::getFileFast(s32 entryId, Abstract::ArchiveHandle::FileInfo &info) const {
     m_handle.open(entryId, info);
     return m_handle.getFileAddress(info);
 }
 
-Archive *Archive::Mount(void *archiveStart) {
+Archive *Archive::FindArchive(void *archiveStart) {
     assert(archiveStart);
-    Archive *archive = nullptr;
-
-    // Check if archive already exists
-    if (!s_archiveList.empty()) {
-        for (size_t i = 0; i < s_archiveList.size(); i++) {
-            if (!s_archiveList[i]) {
-                continue;
-            }
-
-            if (s_archiveList[i]->m_handle.startAddress() == archiveStart) {
-                archive = s_archiveList[i];
-                break;
-            }
+    Archive *foundArchive = nullptr;
+
+    if (!s_listInitialized) {
+        return nullptr;
+    }
+
+    for (auto *archive = s_archiveList.head(); archive; archive = s_archiveList.getNext(archive)) {
+        if (archive->m_handle.startAddress() == archiveStart) {
+            foundArchive = archive;
+            break;
         }
     }
 
-    if (archive) {
+    return foundArchive;
+}
+
+Archive *Archive::Mount(void *archiveStart) {
+    Archive *archive = FindArchive(archiveStart);
+
+    if (!archive) {
+        // Create a new archive and add it to the list
+        archive = new Archive(archiveStart);
+        if (!s_listInitialized) {
+            s_archiveList.init(offsetof(Archive, m_link));
+            s_listInitialized = true;
```

Comment:
If we want to C++ify this some more, this is basically just a singleton. Can we move to that format? Allocate if `nullptr`, delete if count becomes 0?

Reply:
No, as the list is never uninitialized.